src/python_testing/matter_testing_infrastructure/chip/testing/matter_testing.py

```diff
@@ -1451,6 +1455,26 @@ def pics_guard(self, pics_condition: bool):
             self.mark_current_step_skipped()
         return pics_condition
 
+    async def attributes_guard(self, endpoint: int, attribute: str):
```

Comment:
```suggestion
    async def attribute_guard(self, endpoint: int, attribute: ClusterObjects.ClusterAttributeDescriptor):
```
WDYT about wrapping the async part into a asyncio.run so that the caller doesn't need to await this? I think it might be less error prone for authors.

Reply:
I have added the above code and think that is much better than it was previously. 👍

Reply:
I think the problem of implementing the asyncio.run() presviously in place of the await here is that I found it was inside an event loop, so if we ran it from the attribute_guard(), then it complained that we can't do that since asyncio.run() has to be run outside of an active event loop. I had also tried with replacing the await in the test itself, but it also complained there as well.
However, recollecting what you mentioned during our teams call on tuesday, you had mentioned to do the assignment of the self.global_wildcard var after the self.is_commissioning was done, thus I have now moved the assignment of the self.global_wildcard var to the setup_test() instead of the setup_class() in matter testing support module, then put it in the if statement for if not self.is_commissioning and self.runner_hook, this allows us to do the assignment of self.global_wildcard at the start of the test runs and allows us to do it in a asyncio.run().
Please let me know if this will be acceptable.

Reply:
My apologies, while moving the self.global_wildcard var to the setup_test() in matter_testing support module worked on some tests, it unfortunately did not work on some others, you had mentioned this was a concern during the call as well, but I had not noticed this during initial testing. 
I have reverted the code back to a prior version in order to get it to work on all the tests.